cli/op_error.rs

```diff
@@ -0,0 +1,459 @@
+    Self { kind, msg }
+  }
+
+  pub fn not_found(msg: String) -> Self {
+    Self::new(ErrorKind::NotFound, msg)
+  }
+
+  pub fn other(msg: String) -> Self {
+    Self::new(ErrorKind::Other, msg)
+  }
+
+  pub fn type_error(msg: String) -> Self {
+    Self::new(ErrorKind::TypeError, msg)
+  }
+
+  pub fn http(msg: String) -> Self {
+    Self::new(ErrorKind::Http, msg)
+  }
+
+  pub fn uri_error(msg: String) -> Self {
+    Self::new(ErrorKind::URIError, msg)
+  }
+
+  pub fn permission_denied(msg: String) -> OpError {
+    Self::new(ErrorKind::PermissionDenied, msg)
+  }
+
+  pub fn bad_resource() -> OpError {
+    Self::new(ErrorKind::BadResource, "bad resource id".to_string())
+  }
+}
+
+impl Error for OpError {}
+
+impl fmt::Display for OpError {
+  fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+    f.pad(self.msg.as_str())
+  }
+}
+
+impl From<ImportMapError> for OpError {
+  fn from(error: ImportMapError) -> Self {
+    OpError::from(&error)
+  }
+}
+
+impl From<&ImportMapError> for OpError {
+  fn from(error: &ImportMapError) -> Self {
+    Self {
+      kind: ErrorKind::Other,
+      msg: error.to_string(),
+    }
+  }
+}
+
+impl From<ModuleResolutionError> for OpError {
+  fn from(error: ModuleResolutionError) -> Self {
+    OpError::from(&error)
+  }
+}
+
+impl From<&ModuleResolutionError> for OpError {
+  fn from(error: &ModuleResolutionError) -> Self {
+    Self {
+      kind: ErrorKind::URIError,
+      msg: error.to_string(),
+    }
+  }
+}
+
+impl From<VarError> for OpError {
+  fn from(error: VarError) -> Self {
+    OpError::from(&error)
+  }
+}
+
+impl From<&VarError> for OpError {
+  fn from(error: &VarError) -> Self {
+    use VarError::*;
+    let kind = match error {
+      NotPresent => ErrorKind::NotFound,
+      NotUnicode(..) => ErrorKind::InvalidData,
+    };
+
+    Self {
+      kind,
+      msg: error.to_string(),
+    }
+  }
+}
+
+impl From<io::Error> for OpError {
+  fn from(error: io::Error) -> Self {
+    OpError::from(&error)
+  }
+}
+
+impl From<&io::Error> for OpError {
+  fn from(error: &io::Error) -> Self {
+    use io::ErrorKind::*;
+    let kind = match error.kind() {
+      NotFound => ErrorKind::NotFound,
+      PermissionDenied => ErrorKind::PermissionDenied,
+      ConnectionRefused => ErrorKind::ConnectionRefused,
+      ConnectionReset => ErrorKind::ConnectionReset,
+      ConnectionAborted => ErrorKind::ConnectionAborted,
+      NotConnected => ErrorKind::NotConnected,
+      AddrInUse => ErrorKind::AddrInUse,
+      AddrNotAvailable => ErrorKind::AddrNotAvailable,
+      BrokenPipe => ErrorKind::BrokenPipe,
+      AlreadyExists => ErrorKind::AlreadyExists,
+      InvalidInput => ErrorKind::TypeError,
+      InvalidData => ErrorKind::InvalidData,
+      TimedOut => ErrorKind::TimedOut,
+      Interrupted => ErrorKind::Interrupted,
+      WriteZero => ErrorKind::WriteZero,
+      UnexpectedEof => ErrorKind::UnexpectedEof,
+      WouldBlock => unreachable!(),
+      _ => ErrorKind::Other,
```

Comment:
The pass thru should also be unreachable, not map to other. It might be ok to map some std::io::Error types into Other, but it should be explicitly done and justified.